blocks/library/button/index.js

```diff
@@ -101,24 +102,36 @@ class ButtonBlock extends Component {
 							checked={ !! clear }
 							onChange={ this.toggleClear }
 						/>
-						<PanelColor title={ __( 'Background Color' ) } colorValue={ color } >
-							<ColorPalette
-								value={ color }
-								onChange={ ( colorValue ) => setAttributes( { color: colorValue } ) }
-							/>
-						</PanelColor>
-						<PanelColor title={ __( 'Text Color' ) } colorValue={ textColor } >
-							<ColorPalette
-								value={ textColor }
-								onChange={ ( colorValue ) => setAttributes( { textColor: colorValue } ) }
-							/>
-						</PanelColor>
-						{ this.nodeRef && <ContrastCheckerWithFallbackStyles
-							node={ this.nodeRef }
-							textColor={ textColor }
-							backgroundColor={ color }
-							isLargeText={ true }
-						/> }
+						<ColorContext>
```

Comment:
Why did you choose to implement this as a component with children as function. I'm a little bit concerned about this because it's not so obvious for block authors? Do you think it's better to bake these checks inside `PanelColor` and `ColorPalette` instead?

Reply:
Hi @youknowriad, this pattern was used to avoid code duplication https://github.com/WordPress/gutenberg/pull/5570#discussion_r175208006.

Reply:
@youknowriad, I would rather further consolidate those color panels as proposed in https://github.com/WordPress/gutenberg/pull/5570#issuecomment-375655807.

Reply:
Hmm, this would work to:
```js
<PanelColor title={ __( 'Background Color' ) } attributeName="color" value={ color } />
```
😃